deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-service-core.ts

```diff
@@ -0,0 +1,180 @@
+            image: serviceImage,
+            containerName: props.serviceName,
+            command: props.dockerImageCommand,
+            environment: props.environment,
+            portMappings: props.portMappings,
+            logging: LogDrivers.awsLogs({
+                streamPrefix: `${props.serviceName}-logs`,
+                logGroup: serviceLogGroup
+            })
+        });
+        if (props.mountPoints) {
+            serviceContainer.addMountPoints(...props.mountPoints)
+        }
+
+        let cloudMapOptions: CloudMapOptions|undefined = undefined
+        if (props.serviceDiscoveryEnabled) {
+            const namespaceId = StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/cloudMapNamespaceId`)
+            const namespace = PrivateDnsNamespace.fromPrivateDnsNamespaceAttributes(this, "PrivateDNSNamespace", {
+                namespaceName: `migration.${props.stage}.local`,
+                namespaceId: namespaceId,
+                namespaceArn: `arn:aws:servicediscovery:${props.env?.region}:${props.env?.account}:namespace/${namespaceId}`
+            })
+            cloudMapOptions = {
+                name: props.serviceName,
+                cloudMapNamespace: namespace,
+            }
+        }
+
+        const fargateService = new FargateService(this, "ServiceFargateService", {
+            serviceName: `migration-${props.stage}-${props.serviceName}`,
+            cluster: ecsCluster,
+            taskDefinition: serviceTaskDef,
+            assignPublicIp: true,
+            desiredCount: props.taskInstanceCount ? props.taskInstanceCount : 1,
+            enableExecuteCommand: true,
+            securityGroups: props.securityGroups,
+            // This should be confirmed to be a requirement for Service Connect communication, otherwise be Private
+            vpcSubnets: props.vpc.selectSubnets({subnetType: SubnetType.PUBLIC}),
+            serviceConnectConfiguration: {
+                namespace: `migration.${props.stage}.local`,
+                services: props.serviceConnectServices ? props.serviceConnectServices : undefined,
+                logDriver: LogDrivers.awsLogs({
+                    streamPrefix: "service-connect-logs",
+                    logGroup: serviceLogGroup
+                })
+            },
+            cloudMapOptions: cloudMapOptions
+        });
+        // Use CDK escape hatch to modify the underlying CFN for the generated AWS::ServiceDiscovery::Service to allow
+        // multiple DnsRecords. GitHub issue can be found here: https://github.com/aws/aws-cdk/issues/18894
+        if (props.serviceDiscoveryEnabled) {
+            const multipleDnsRecords = {
+                DnsRecords: [
+                    {
+                        TTL: 10,
+                        Type: "A"
+                    },
+                    {
+                        TTL: 10,
+                        Type: "SRV"
+                    }
+                ]
```

Comment:
Why do we need different kinds of records?  Where are we using SRVs?

Reply:
I don't understand the specific details of why this is required, but without both an A and SRV record I was not able to communicate via a specific port between our Migration services without Service Connect.